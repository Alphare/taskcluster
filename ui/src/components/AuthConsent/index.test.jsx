import React from 'react'
import { shallow } from 'enzyme'

import AuthConsent from './index'

import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
const history = createMemoryHistory({})

it('should render AuthConsent page', () => {
  const cmp = shallow(<Router history={history}>
    <AuthConsent
      transactionID="transactionID"
      registeredClientId="registeredClientId"
      clientId="clientId"
      onExpirationChange={jest.fn()}
      onInputChange={jest.fn()}
      onScopesChange={jest.fn()}
      formData={{
        expires: new Date('2022-02-02T12:00:00.000Z'),
        description: 'description',
        scopes: ['scopes'],
      }}
    />
  </Router>)

  expect(cmp).toBeDefined()
  expect(cmp.html()).toContain('input')
  expect(cmp).toMatchSnapshot()
})