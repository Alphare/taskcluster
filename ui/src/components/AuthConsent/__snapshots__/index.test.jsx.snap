// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AuthConsent page 1`] = `
<ContextProvider
  value={
    Object {
      "history": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "gjgxjz",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "gjgxjz",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "location": Object {
        "hash": "",
        "key": "gjgxjz",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "match": Object {
        "isExact": true,
        "params": Object {},
        "path": "/",
        "url": "/",
      },
      "staticContext": undefined,
    }
  }
>
  <ContextProvider
    value={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "gjgxjz",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "gjgxjz",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(WithStyles(AuthConsent))
      clientId="clientId"
      formData={
        Object {
          "description": "description",
          "expires": 2022-02-02T12:00:00.000Z,
          "scopes": Array [
            "scopes",
          ],
        }
      }
      onExpirationChange={[MockFunction]}
      onInputChange={[MockFunction]}
      onScopesChange={[MockFunction]}
      registeredClientId="registeredClientId"
      transactionID="transactionID"
    />
  </ContextProvider>
</ContextProvider>
`;
